### PR TITLE
VPN Server Development

### PR DESCRIPTION
**Pull request by SunithaGudisagarIBM**
Community Note
Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request
Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TEST=./ibm/service/vpc TESTARGS='-run=TestAccIBMIsVPNServerBasic'
$ make testacc TEST=./ibm/service/vpc TESTARGS='-run=TestAccIBMIsVPNServerRouteBasic'
$ make testacc TEST=./ibm/service/vpc TESTARGS='-run=TestAccIBMIsVPNServersDataSourceBasic'
$ make testacc TEST=./ibm/service/vpc TESTARGS='-run=TestAccIBMIsVPNServerDataSourceBasic'
$ make testacc TEST=./ibm/service/vpc TESTARGS='-run=TestAccIBMIsVPNServerClientConfigDataSourceBasic'
$ make testacc TEST=./ibm/service/vpc TESTARGS='-run=TestAccIBMIsVPNServerRouteDataSourceBasic'
$ make testacc TEST=./ibm/service/vpc TESTARGS='-run=TestAccIBMIsVPNServerRoutesDataSourceBasic'

...
```

### CODE REVIEW

File: ibm/service/vpc/resource_ibm_is_vpn_server_client.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+)
+
+func ResourceIBMIsVPNServerClient() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIBMIsVPNServerClientDisconnect,
```
Comment (hkantare) [inline ibm/service/vpc/resource_ibm_is_vpn_server_client.go:20]: All Create, Read and Update handles same method?

File: ibm/service/vpc/data_source_ibm_is_vpn_server_client_config.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+			_, err = f.WriteString(configStr)
+		}
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("[ERROR] Error Saving VPNServerClientConfiguration Result: %s", err))
+
+		}
+		log.Printf("OpenVPN client configuration was saved to {{.%s}}", fileName)
+	}
+
+	if err = d.Set("vpn_server_client_configuration", *result); err != nil {
```
[the remark this answers is not in the document]
Comment (SunithaGudisagarIBM) [reply ibm/service/vpc/data_source_ibm_is_vpn_server_client_config.go:79]: https://pages.github.ibm.com/vpc/vpc-spec-artifacts/branch/master/public/2022-03-29.html#/VPN%20servers/get_vpn_server_client_configuration

It stores details of the vpn server client configuration

File: ibm/service/vpc/resource_ibm_is_vpn_server.go [hunk 3]
```diff
@@ -0,0 +1,905 @@
...
+						"identity_provider": &schema.Schema{
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validate.ValidateAllowedStringValues([]string{"iam"}),
+							Description:  "The type of identity provider to be used by the VPN client.- `iam`: IBM identity and access managementThe enumerated values for this property are expected to expand in the future. When processing this property, check for and log unknown values. Optionally halt processing and surface the error, or bypass the route on which the unexpected property value was encountered.",
+						},
+						"client_ca_crn": &schema.Schema{
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "The CRN for this certificate instance.",
```
[the remark this answers is not in the document]
Comment (SunithaGudisagarIBM) [reply ibm/service/vpc/resource_ibm_is_vpn_server.go:76]: corrected

File: ibm/service/vpc/resource_ibm_is_vpn_server_client.go (hunk 1 above)
Comment (SunithaGudisagarIBM) [reply ibm/service/vpc/resource_ibm_is_vpn_server_client.go:20]: Yes,

**Review by hkantare — APPROVED** [review]

**Pull request merged**